### PR TITLE
Multi extension support

### PR DESCRIPTION
**Pull request by phated**
@sttk I've had this as a WIP for quite awhile but didn't submit it because the tests were **very** out-of-date and I wanted to get things passing before I sent it.

Can you review and determine if this is a proper solution for #26 and #34?  I think I've properly accounted for all the weirdness around multiple extensions but another set of eyes is needed.

### CODE REVIEW

**Comment by phated** [comment]
I'd also like to get this landed before I rebase the project to inline with the other gulp repos.

File: index.js [hunk 1]
```diff
@@ -9,18 +9,30 @@ exports.prepare = function (extensions, filepath, cwd, nothrow) {
...
+  var usedExtension;
+  if (exts) {
+    exts.some(function(ext) {
+      usedExtension = ext;
+      config = normalize(extensions[ext]);
+      return !!config;
+    });
+  }
+
+  if(Object.keys(require.extensions).indexOf(usedExtension) !== -1) {
```
[the remark this answers is not in the document]
Comment (phated) [reply index.js]: I think it's fine to keep like this.

File: lib/extension.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
 const path = require('path');
 
-const EXTRE = /^[.]?[^.]+([.].*)$/;
+const EXTRE = /\.[^.]+/g;
+const LONGEXTRE = /^[.]?[^.]+([.].+)$/;
```
[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk yeah, I think your regexp is better than mine.  Let me come up with some tests.

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk actually, it seems the `LONGEXTRE` needs to capture from the first `.` to the end of line, including future `.`s.

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk I can see what you are trying to achieve.  What about using this RegExp: `/^[.]?[^.]+([.].+[^.])$/` (visualized at [regexper](https://regexper.com/#%2F%5E%5B.%5D%3F%5B%5E.%5D%2B%28%5B.%5D.%2B%5B%5E.%5D%29%24%2F))

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: I've made the above changes that I suggested and added a test.  Let me know what you think.

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: Oh, I mistook, too. And your change is not correct yet.

```js
/^[.]?[^.]+([.].+[^.])$/.exec('aaa.bbb..js'); // => [ 'aaa.bbb..js', '.bbb..js' ...]

'.bbb..js'.match(/\.[^.]+/g); // => [ '.bbb', '.js' ]
```

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: RegExp is hard!  Do we need to change EXTRE instead of LONGEXTRE then?

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk what do you think is the correct results for `aaa.bbb..js`?  Should the extension be considered `bbb..js`?  Or should it just be considered invalid?

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: It may be easy to use `String#split('.')`

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk but then we have an empty string between the double-dot and you get the same results.  So what is the desired result?

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: @phated Yes, we can have an empty string between double-dot or by last dot, and we can control if they are error or not.

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk is this correct?
```js
expect(extension('file.babel..js')).to.equal(['babel..js']);
```

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: I tried that and it seems to work but here are some cases that we need to define:
```js
extension('file..babel.js')

extension('file.babel..js')

extension('file.something..babel.js')
```

I think some of these should return proper extensions, even though there's weird double-dot syntax inside the basename.

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: @phated I think it is more important to match the results by `LONGEXTRE` and `EXTRE` than to allow double-dot.
But I found the above regexp which does not allow double-dot.

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk your above RegExp is pretty good but it completely rejects ANY path that has double dots.  I think that is incorrect.

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: > I think some of these should return proper extensions, even though there's weird double-dot syntax inside the basename.

Sure. We should change `EXTRE` to support double-dot.

[the remark this answers is not in the document]
Comment (sttk) [reply lib/extension.js]: @phated Which should we adopt for `EXTRE`?

1. `'.bbb.ccc..js'.match(/\.+[^.]+/g); // => [ '.bbb', '.ccc', '..js' ]`
2. `'.bbb.ccc..js'.match(/\.[^.]*/g); // => [ '.bbb', '.ccc', '.', '.js' ]`

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk I think 2 is correct.

[the remark this answers is not in the document]
Comment (phated) [reply lib/extension.js]: @sttk I just pushed changes that use your suggestion (2) and included tests.  Let me know if you think that solves the issue.

**Comment by phated** [comment]
Thanks for reviewing @sttk!!!

**Comment by sttk** [comment]
@phated Great work!

**Pull request merged**